### PR TITLE
Fetch resources in create-report action

### PR DESCRIPTION
**Pull request by dehall**
Fetches resources as specified by the `PlanDefinition.action.input` field on the `create-report` action. Note that this should work when the trigger is a Patient or an Encounter, not sure about other resource types such as Condition yet. 

I used the attached encounter resource, pointing to all our internal FHIR servers, for testing:
[mitre_sample_enc.json.txt](https://github.com/mcode/medmorph-backend/files/6462135/mitre_sample_enc.json.txt)

### CODE REVIEW

File: src/utils/actions.js [hunk 1]
```diff
@@ -66,8 +66,83 @@ const baseIgActions = {
...
+            // ASSUMPTION: if given an encounter context,
+            //   we only care about that one, not all encounters
+            continue;
+          } else {
+            // assume we have context.patient
+            params.push(`subject=Patient/${context.patient.id}`);
+          }
+        } else {
+          // TODO: logic to check which resources actually have these fields?
+          // will be necessary if we get away from the simple US core types
```
Comment (blangley28) [inline src/utils/actions.js:110]: This is a good point. For the connectathon it looks like this is fine but definitely something we should look into (and possibly create a new task for the backlog if required)

File: src/utils/actions.js [hunk 2]
```diff
@@ -66,8 +66,83 @@ const baseIgActions = {
...
+        //   // TODO: what if the server doesn't include profiles or support the search filter?
+        //   params.push(`_profile=${input.profile[0]}`);
+        //   // note also specifying profile may not be ideal
+
+        // eslint-disable-next-line max-len
+        //   // see: https://chat.fhir.org/#narrow/stream/179252-IG-creation/topic/Requiring.20meta.2Eprofile.20be.20populated
+        // }
+
+        const query = resourceType + '?' + params.join('&');
+        const bundle = (await readFromEHR(query)).data;
```
Comment (blangley28) [inline src/utils/actions.js:137]: Instead of using `await` in a loop, keep a list of promises then and then call `.all`

**Review by blangley28 — CHANGES_REQUESTED** [review]

File: src/utils/actions.js (hunk 1 above)
Comment (dehall) [reply src/utils/actions.js:110]: Created MEDMORPH-121 for this

File: src/utils/reporting_workflow.js [hunk 3]
```diff
@@ -275,6 +275,11 @@ async function executeWorkflow(context) {
     const action = planDef.action.find(a => a.id === currentActionId);
     context.action = action;
 
+    if (action.action) {
+      // connectathon hack - merge everything from the nested action into this one
+      Object.assign(action, action.action[0]);
+    }
+
```
[the remark this answers is not in the document]
Comment (dehall) [reply src/utils/reporting_workflow.js:282]: Agree in spirit, but I don't want to be caught unprepared if for any reason the un-fixed PlanDef is still the one they use. I'd rather delete this once the change is published. And in the meantime, this won't break anything else that doesn't use a nested action

**Review by blangley28 — APPROVED** [review]
LGTM, testing with an Observation and it was successfully added to the content bundle. One quick thought - we only add to the report to the completed reports table on the "complete-report" action. For the connectathon it would be nice to have this populated. We could either:
1. Insert it on submit and complete reporting
2. Ask Dragon to add "complete-reporting" action to the PlanDefinition. I'm not sure what other people do during this step but it kind of sounded like every PlanDefinition should end with this anyway.

**Comment by dehall** [comment]
Yeah I noticed that too. Based on the list of action codes it would seem like that one should always be last, but it's not specifically called out as necessary anywhere: https://build.fhir.org/ig/HL7/fhir-medmorph/provisioning.html 

It's a few more clicks, but for something like the connectathon where we probably won't have dozens of reports, I'm fine with just clicking through the Reporting table to find the submitted bundle

**Review by danlee1025 — APPROVED** [review]

**Pull request merged**